src/main/java/services/RestoreFactory.java

```diff
@@ -487,7 +517,17 @@ public String getUserRestoreUrl() {
         if( asUsername != null) {
             builder.append("&as=").append(asUsername).append("@").append(domain).append(".commcarehq.org");
         }
-        return builder.toString();
+        String restoreUrl = builder.toString();
+        HttpHeaders headers;
+        if (getHqAuth() == null) {
+            // Need to do HMAC auth
+            headers = getHmacHeaders(restoreUrl);
```

Comment:
won't this fail since `getHmacHeaders` can only be called for requests with `requestValidatedWithHMAC = True`?

Reply:
@snopoke My understanding is that now SMS will use HMAC auth for all requests, so some SMS requests will use HMAC auth for the standard restore auth in this block.